src/sql/src/plan/func.rs

```diff
@@ -694,6 +707,7 @@ impl<'a> ArgImplementationMatcher<'a> {
             ParamType::Plain(List(..)) if matches!(arg_type, List(..)) => return Ok(arg),
             ParamType::Plain(s) => CastTo::Implicit(s.clone()),
             ParamType::Any => return Ok(arg),
+            ParamType::ArrayAny => CastTo::Explicit(Array(Box::new(String))),
```

Comment:
For sure! Done. Left it as a TODO at the top of the method though because I think the same reasoning applies to the `coerce_to` logic in addition to the `cast_to`.